### PR TITLE
adding security api reference #887

### PR DESCRIPTION
**Pull request by AntonEliatra**
### Description
Adding references for security APIs

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/887

### Version
_all._

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by AntonEliatra** [comment]
We can move this to https://opensearch.org/docs/latest/security/access-control/api/
and add examples of the calls, but that section is under "Access Control", although some of the APIs listed on that page have nothing to do with access control

**Comment by hdhalter** [comment]
@AntonEliatra - Are you referring to the content under [this](https://opensearch.org/docs/latest/security/access-control/api/) page or in the [issue](https://github.com/opensearch-project/documentation-website/issues/887)? 

@Naarcha-AWS

**Review by scrawfor99 — APPROVED** [review]
Looks good

File: _api-reference/security-apis.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+layout: default
+title: Security APIs
+nav_order: 84
+---
+
+# Security APIs
+
+Security APIs provide information which can be very useful in troubleshooting connection issues and overall configuration.
```
Comment (natebower) [inline _api-reference/security-apis.md:9]:
```suggestion
Security APIs provide information that can be very useful in troubleshooting connection and configuration issues.
```

File: _api-reference/security-apis.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+API | Method | Description
+:--- | :--- | :---
+`/_plugins/_security/whoami` | GET/POST | Returns basic details about the logged in user.
```
Comment (natebower) [inline _api-reference/security-apis.md:13]:
```suggestion
`/_plugins/_security/whoami` | GET/POST | Returns basic details about the logged-in user.
```

File: _api-reference/security-apis.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+`/_opendistro/_security/sslinfo` | GET | Returns details about the SSL connection when using certificate authentication.
+`/_plugins/_security/api/permissionsinfo` | GET | Returns permission details for the logged in user.
```
Comment (natebower) [inline _api-reference/security-apis.md:15]:
```suggestion
`/_plugins/_security/api/permissionsinfo` | GET | Returns permission details for the logged-in user.
```

File: _api-reference/security-apis.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+`/_plugins/_security/authinfo` | GET/POST | Returns the backend roles and OpenSearch roles mapped to the logged in user.
```
Comment (natebower) [inline _api-reference/security-apis.md:16]:
```suggestion
`/_plugins/_security/authinfo` | GET/POST | Returns the backend roles and OpenSearch roles mapped to the logged-in user.
```

File: _api-reference/security-apis.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+`/_plugins/_security/api/ssl/certs` | GET | Displays details and expiration of the certificates used in OpenSearch HTTP and transport communication layers. Can only be called with by users with the `superadmin` certificate.
```
Comment (natebower) [inline _api-reference/security-apis.md:17]:
```suggestion
`/_plugins/_security/api/ssl/certs` | GET | Displays the details and expiration dates of the certificates used on the OpenSearch HTTP and transport communication layers. Can only be called by users with the `superadmin` certificate.
```

File: _api-reference/security-apis.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+`/_plugins/_security/api/ssl/transport/reloadcerts` | PUT | Reload the certificates on the `transport` layer. For more information, see [Reload TLS certificates on the transport layer]({{site.url}}{{site.baseurl}}/security/configuration/tls/#reload-tls-certificates-on-the-transport-layer).
```
Comment (natebower) [inline _api-reference/security-apis.md:18]:
```suggestion
`/_plugins/_security/api/ssl/transport/reloadcerts` | PUT | Reloads the certificates on the `transport` layer. For more information, see [Reload TLS certificates on the transport layer]({{site.url}}{{site.baseurl}}/security/configuration/tls/#reload-tls-certificates-on-the-transport-layer).
```

File: _api-reference/security-apis.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+`/_plugins/_security/api/ssl/http/reloadcerts` | PUT | Reload the certificates on the `http` layer. For more information, see [Reload TLS certificates on the http layer]({{site.url}}{{site.baseurl}}/security/configuration/tls/#reload-tls-certificates-on-the-http-layer).
```
Comment (natebower) [inline _api-reference/security-apis.md:19]:
```suggestion
`/_plugins/_security/api/ssl/http/reloadcerts` | PUT | Reloads the certificates on the `http` layer. For more information, see [Reload TLS certificates on the http layer]({{site.url}}{{site.baseurl}}/security/configuration/tls/#reload-tls-certificates-on-the-http-layer).
```

**Review by natebower — COMMENTED** [review]
@Naarcha-AWS @AntonEliatra Please see my changes and let me know if you have any questions. Thanks!

**Review by Naarcha-AWS — APPROVED** [review]

**Pull request merged**